# Задание для DS по разработке PD модели

## Необходиом разработать модель предсказания дефолта. Даны финансовые данные юридических лиц (выручка, активы, пассивы и т.д). 

### Импортируем библиотеки

In [18]:
import pandas as pd
from sklearn.ensemble import GradientBoostingClassifier as not_xgb
from sklearn.metrics import f1_score
%matplotlib inline
from matplotlib import pyplot as plt

### Подгружаем данные

In [16]:
!wget https://raw.githubusercontent.com/BKHV/ds-test/master/PD-data.csv

/bin/sh: wget: command not found


In [17]:
df = pd.read_csv('PD-data.csv', sep=';')

Размеры выборок

In [7]:
print(df.shape)

(32595, 37)


Количество разных типов фичей

In [9]:
df.dtypes.value_counts()

float64    25
int64      11
object      1
dtype: int64

Уровень дефолт-рейта в выборке = 6,7219%

In [10]:
df.default_12m.value_counts(normalize=True)

0    0.932781
1    0.067219
Name: default_12m, dtype: float64

In [12]:
df.record_id.nunique()

32595

32595 уникальных наблюдений

### Строим модель

Начиная с ячейки ниже текущей просим произвести построение модели предскзаания вероятности дефолта. Итоговое решение прислать сотрудникам Банка в формате Jupiter Notebook файла